modules/xmpp/Caps.js

```diff
@@ -187,28 +187,35 @@ export default class Caps extends Listenable {
     _getDiscoInfo(jid, node, timeout) {
         return new Promise((resolve, reject) =>
             this.disco.info(jid, node, response => {
-                const features = new Set();
-                const identities = new Set();
-
-                $(response)
-                    .find('>query>feature')
-                    .each(
-                        (_, el) => features.add(el.getAttribute('var')));
-                $(response)
-                    .find('>query>identity')
-                    .each(
-                        (_, el) => identities.add({
-                            type: el.getAttribute('type'),
-                            name: el.getAttribute('name'),
-                            category: el.getAttribute('category')
-                        }));
-                resolve({
-                    features,
-                    identities });
+                resolve(this.parseDiscoInfo(response));
```

Comment:
I'm not sure I understand the js well, but does `parseDiscoInfo` need to be defined on the object? It looks like it only operates on its `node` parameter.

Reply:
It is defined in the object and same method is used in another place ... an option is to move it in util folder so can be used from both places ... but we will extract xmpp only and caps only functionality in another place ... so I don't see the point of that.

Reply:
I fixed it with the latest commit.